### PR TITLE
Update XNNPACK Version

### PR DESCRIPTION
**Pull request by mcr229**
Updating XNNPACK Version to 4ea82e595b36106653175dcb04b2aa532660d0d8

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @albanD

submodule update

### CODE REVIEW

**Review by digantdesai — APPROVED** [review]

**Review by digantdesai — APPROVED** [review]

File: cmake/Dependencies.cmake [hunk 1]
```diff
@@ -516,6 +516,9 @@ if(USE_XNNPACK AND NOT USE_SYSTEM_XNNPACK)
     # Disable I8MM For CI since clang 9 does not support neon i8mm.
     set(XNNPACK_ENABLE_ARM_I8MM OFF CACHE BOOL "")
 
+    # Disable avxvnni int8
```
[the remark this answers is not in the document]
Comment (mcr229) [reply cmake/Dependencies.cmake:519]: this was breaking some CI so I just disabled. I forgot which one and why though :(

File: third_party/xnnpack_src_defs.bzl [hunk 2]
```diff
@@ -2,406 +2,1435 @@
 Auto-generated by generate-wrappers.py script. Do not modify
 """
 
-PROD_ARMSIMD32_MICROKERNEL_SRCS = [
-    "XNNPACK/src/amalgam/gen/armsimd32.c",
+XNNPACK_SRCS = [
```
[the remark this answers is not in the document]
Comment (mcr229) [reply third_party/xnnpack_src_defs.bzl:5]: yea I think I accidentally still ran the gen_wrappers script which edited these. These are no longer used even though they are modified. I will delete everything from that in a follow up diff.

**Review by huydhn — APPROVED** [review]

**Comment by huydhn** [comment]
I'm pretty sure that the https://github.com/pytorch/pytorch/actions/runs/11833440031/job/33005664656 is unrelated and it looks like an infra issue with the runner

**Pull request closed without merging**